src/main/blackbox/blackbox_io.c

```diff
@@ -145,12 +145,24 @@ void blackboxWriteUnsignedVB(uint32_t value)
 }
 
 /**
+ * ZigZag encoding maps all values of a signed integer into those of an unsigned integer in such
+ * a way that numbers of small absolute value correspond to small integers in the result.
+ *
+ * (Compared to just casting a signed to an unsigned which creates huge resulting numbers for
+ * small negative integers).
+ */
+static uint32_t zigzagEncode(int32_t value)
+{
+    return (uint32_t)((value << 1) ^ (value >> 31));
+}
+
```

Comment:
Not sure if you can use the following for test (I found on internet) 
( value >> 1 ) ^ ( ~( value & 1 ) + 1 )